src/main/groovy/val/Idator.groovy

```diff
@@ -22,6 +79,280 @@ class Idator {
      */
     //TODO: allow per instance extension
     void registerChecker(String name, Closure closure) {
-        DefinitionCheck.metaClass[name] = closure
+        extraChecks[name] = closure
```

Comment:
This was switched around to use the `methodMissing` beneath since I was having issues with the metaClass and this is also more in line with how extension could be done in general: having a delegate which is then called by the meta layer. Some variation on this approach should also allow IntelliJ to pick up the custom defined methods, likely by using more of a mixin object approach than individual methods.